Calendar/CalendarViewController.m

```diff
@@ -126,15 +137,87 @@ - (void) fetchProfesionals{
+                return event.calendar.type == EKCalendarTypeLocal || event.calendar.type == EKCalendarTypeCalDAV;
+            }]];
+            
+            dispatch_async(dispatch_get_main_queue(), ^{
+                if (!weakSelf) return;
+                weakSelf.calendarEvents = [[NSMutableArray alloc] initWithArray: events];
+                [weakSelf.calendar reloadData];
+            });
+        }else{
+            //Alert
+            UIAlertController *alertController = [UIAlertController alertControllerWithTitle:@"Permission Error" message:@"Permission fo calendar is required for fetching events" preferredStyle:UIAlertControllerStyleAlert];
+            [alertController addAction:[UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleCancel handler:nil]];
+            [self presentViewController:alertController animated:YES completion:nil];
+        }
+    }];
+   
+}
+ 
+
 // Returns the number of events per date using orderEvents
 - (NSInteger)calendar:(FSCalendar *)calendar numberOfEventsForDate:(NSDate *)date{
     if ([self.distinctEvents containsObject:[self.dateFormatter stringFromDate:date]]){
+        int eventCount = 0;
         NSArray * events = [self.orderEvents objectForKey:[self.dateFormatter stringFromDate:date]];
-        return events.count;
+        eventCount = (int)events.count;
+        
+        for (EKEvent* event in self.calendarEvents){
+            NSString * startDateEvent = (NSString*)[self.dateFormatter stringFromDate:event.startDate];
+            if ([startDateEvent isEqual:[self.dateFormatter stringFromDate:date]]){
+                eventCount += 1;
+            }
+        }
+        return eventCount;
+    }else{
+        for (EKEvent* event in self.calendarEvents){
+            NSString * startDateEvent = (NSString*)[self.dateFormatter stringFromDate:event.startDate];
+            if ([startDateEvent isEqual:[self.dateFormatter stringFromDate:date]]){
+                return 1;
+            }
+        }
     }
     return 0;
 }
 
+-(NSArray *)calendar: (FSCalendar *)calendar appearance:(FSCalendarAppearance *)appearance eventDefaultColorsForDate:(NSDate *)date{
+    
+    if ([self.distinctEvents containsObject:[self.dateFormatter stringFromDate:date]]){
+        int eventCount = 0;
+        NSArray * events = [self.orderEvents objectForKey:[self.dateFormatter stringFromDate:date]];
+        eventCount = (int)events.count;
+        
+        for (EKEvent* event in self.calendarEvents){
+            NSString * startDateEvent = (NSString*)[self.dateFormatter stringFromDate:event.startDate];
+            if ([startDateEvent isEqual:[self.dateFormatter stringFromDate:date]]){
+                return @[appearance.eventDefaultColor, [UIColor blackColor]];
+            }
+        }
+        return @[appearance.eventDefaultColor];;
+    }else{
+        for (EKEvent* event in self.calendarEvents){
```

Comment:
Could you also explain what this else logic is doing?

Reply:
Sure, I will comment a description in the code. But I'm basically checking to see if the event is within the start/end date range which in this case spans form 2022 to 2024. Additionally, Im checking to see if the event was created on the app or is a personal event to color code it accordingly.